### ISSUE TITLE
Save session periodically

### ISSUE THREAD

**Issue by PeterBenc** [nlpsuge/gnome-shell-extension-another-window-session-manager#60]
Setup: 
 - I don't use multiple sessions, what I want is to be able to restart my pc and have everything restored, that's it.

Issue: 
 - I find myself clicking the "Save current session" multiple times during the day because it already happened a couple of times that I forgot to do that and the last used session was too old. 
 
 Proposal: 
  - Have the option to set periodical savings of the current session. e.g. every 5 minutes the current session gets saved. 
  - There are already propositions to save before the shutdown, but I suppose that won't be that easy also considering emergency shutdowns and this is a very simple fix for that. Normal shutdown or not, your sessions would be saved some time before the shutdown because it is saved periodically

**Comment by nlpsuge** [comment]
Hi, thanks for your idea.

I had added an option `Stash and restore` under `Restore sessions` tab, but currently this option is only useful when **gnome-shell restarts** via `Alt+F2 type r` or `killall -3 gnome-shell`. 

And when `Stash and restore` is enable, this extension only saves the window-state-changed windows, such as changes of position, title, size, workspace, monitor, etc, so it won't occupy too many resources compared to saving all windows periodically. 

My plan is at least extending the above feature to emergency shutdowns. So when the OS restarts, a dialog pops up to let user to decide to restore the previous apps and windows or not, just like Firefox browser does. 

I think this will take time to implement, for I don't have much time recently.

Related to #9

**Comment by areksobiczewski** [comment]
I just wanted to +1 the OP's request.
There are more of us who use the AWSM plugin mostly for handling OS restarts only.

Once in a while gnome-shell goes thru a period of unstability* regarding its updates. During such a period users are loosing work! The AWSM plugin could prevent that 100%. Other than lacking an emergency shutdowns options the plugin is perfect perfect. It's a no. 1 contender for an emergency shutdowns option for GNOME.
Another case I went thru recently is with configuring bluetooth wireless LDAC headphones (or any other hardware wise changes). Due to some misconfigurations I experienced multiple kernel panics with bluetooth, which could have been completely "prevented" with the AWSM plugin saving my work opened in multiple windows / desktops. It would make my life so much easier.

I hope that @nlpsuge, you are going to find some time to implement that feature. I'd consider that new option a killer feature for the plugin TBO.
Other than that, I'd like to thank you for your past contributions on the AWSM. You're awesome!

*) Recently that happened with the gnome-shell-42.4 as mentioned here:
https://www.reddit.com/r/gnome/comments/xn0t7c/has_gnome_recently_become_very_unstable_for_you/
